src/Notes/Collection/AbstractCompleteOnboarding.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Pinterest for WooCommerce Abstract Marketing Note class.
+ *
+ * @package Pinterest_For_WooCommerce/Classes/
+ * @version x.x.x
+ */
+
+namespace Automattic\WooCommerce\Pinterest\Notes\Collection;
+
+use Automattic\WooCommerce\Admin\Notes\Note;
+use Automattic\WooCommerce\Pinterest\Notes\MarketingNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AbstractCompleteOnboarding.
+ *
+ * Base class for a set of onboarding reminders.
+ *
+ * @since x.x.x
+ */
+abstract class AbstractCompleteOnboarding extends AbstractNote {
+
+	// Timestamp option marking the moment we start to count time.
+	const INIT_TIMESTAMP = MarketingNotifications::OPTIONS_PREFIX . '-init-timestamp';
+
+	/**
+	 * Should the note be added to the inbox.
+	 *
+	 * @since x.x.x
+	 * @return boolean
+	 */
+	public function should_be_added(): bool {
+		if ( Pinterest_For_Woocommerce()::is_setup_complete() ) {
+			return false;
+		}
+
+		if ( self::note_exists() ) {
+			return false;
+		}
+
+		// Are we there yet?
+		if ( time() < ( DAY_IN_SECONDS * static::DELAY + $this->get_init_timestamp() ) ) {
+			return false;
+		}
+
+		// Check if we have enough orders to proceed.
+		$args = array(
+			'limit'  => 5,
+			'status' => array( 'wc-completed' ),
+			'return' => 'ids',
+		);
+
+		$orders_ids = wc_get_orders( $args );
+		if ( 5 > count( $orders_ids ) ) {
+			return false;
+		}
+
+		// All preconditions are met, we can send the note.
+		return true;
+	}
+
+	/**
+	 * Add button to Pinterest For WooCommerce landing page.
+	 *
+	 * @since x.x.x
+	 * @param Note $note Note to which we add an action.
+	 */
+	protected function add_action( $note ): void {
+		$note->add_action(
+			'goto-pinterest-landing',
+			__( 'Complete setup', 'pinterest-for-woocommerce' ),
+			wc_admin_url( '&path=/pinterest/landing' )
+		);
+	}
+
+	/**
+	 * Check if the init timestamp exist.
+	 * Initialize if not.
+	 *
+	 * @since x.x.x
+	 * @return int Initialization timestamp.
+	 */
+	protected function get_init_timestamp(): int {
+		$timestamp = get_option( self::INIT_TIMESTAMP );
+		if ( false !== $timestamp ) {
+			// Timestamp already init, return.
+			return (int) $timestamp;
+		}
+
+		// Timestamp not initialized, create a new one.
+		$timestamp = time();
+		add_option( self::INIT_TIMESTAMP, $timestamp );
+		return $timestamp;
+	}
```

Comment:
I know this is very opinionated, but my thought was that this function would be clearer if we initialize within the if statement and have only one return. Which in my opinion makes the extra comment lines unnecessary since we can describe what it does in the function comment.
```suggestion
	protected function get_init_timestamp(): int {
		$timestamp = get_option( self::INIT_TIMESTAMP );
		if ( false === $timestamp ) {
			$timestamp = time();
			add_option( self::INIT_TIMESTAMP, $timestamp );
		}

		return (int) $timestamp;
	}
```